loader.go

```diff
@@ -16,20 +18,79 @@ import (
 type LoaderEntry struct {
 	ID        float64   `json:"id"`        // Loader ID
 	Name      string    `json:"name"`      // Loader name
+	Prefix    string    `json:"prefix"`    // Loader key prefix
 	Key       string    `json:"key"`       // Loader key (only populated during creation)
 	AgentName string    `json:"agentname"` // Loader environment, agent name
 	LastSeen  time.Time `json:"lastseen"`  // Last time loader was used
 	Enabled   bool      `json:"enabled"`   // Loader entry is active
 }
 
-func (le *LoaderEntry) Validate() error {
+func (le *LoaderEntry) Validate() (err error) {
+	if le.Key != "" {
+		err = ValidateLoaderPrefixAndKey(le.Prefix + le.Key)
+	}
 	return nil
 }
 
+// Generate a new loader prefix value
+func GenerateLoaderPrefix() string {
+	return RandLoaderKeyString(8)
+}
+
+// Generate a new loader key value
+func GenerateLoaderKey() string {
+	return RandLoaderKeyString(32)
+}
+
+// RandLoaderKeyString is used for prefix and key generation, and just
+// returns a random string consisting of alphanumeric characters of
+// length characters long
+func RandLoaderKeyString(length int) string {
+	ret := make([]byte, length)
+	lset := []byte("abcdefghijklmnopqrstuvwxyzABCDEFCHIJKLMNOPQRSTUVWXYZ0123456789")
+	r := mrand.New(mrand.NewSource(time.Now().UnixNano()))
+	for i := 0; i < len(ret); i++ {
+		ret[i] = lset[r.Int()%len(lset)]
+	}
+	return string(ret[:len(ret)])
```

Comment:
Note this is also why we have only a minimum here: https://github.com/mozilla/mig/blob/7751771158e06dc23220f533fbcdecf3c9475035/mig-agent/context.go#L317, since the resulting string length is variable.